### PR TITLE
Add support for conv2d in SPDZ 

### PR DESCRIPTION
**Pull request by LaRiffle**
## Description
Add conv2d in SPDZ + tests

### CODE REVIEW

File: src/sympc/protocol/beaver/beaver.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 
 def _get_triples(
-    op_str: str, nr_parties: int, a_shape: Tuple[int], b_shape: Tuple[int]
+    op_str: str, nr_parties: int, a_shape: Tuple[int], b_shape: Tuple[int], **kwargs
```
Comment (gmuraru) [inline src/sympc/protocol/beaver/beaver.py:28]: Q: Could you add type hint for ```kwargs```: ```Dict[Any, Any```

File: src/sympc/protocol/beaver/beaver.py [hunk 2]
```diff
@@ -171,6 +174,53 @@ def matmul_store_get(
...
 
 
+""" Beaver Operations defined for Convolution 2D """
+
+
+@register_primitive_generator("beaver_conv2d")
+def get_triples_conv2d(
+    *args: List[Any], **kwargs: Dict[Any, Any]
+) -> Tuple[List[ShareTensor], List[ShareTensor], List[ShareTensor]]:
+    """Get the beaver triples for the mul operation."""
```
Comment (gmuraru) [inline src/sympc/protocol/beaver/beaver.py:184]: TYPO: For the conv2d

File: src/sympc/protocol/spdz/spdz.py [hunk 3]
```diff
@@ -19,13 +19,13 @@
...
 
-EXPECTED_OPS = {"mul", "matmul"}
+EXPECTED_OPS = {"mul", "matmul", "conv2d"}
 
 
 """ Functions that are executed at the orchestrator """
 
 
-def mul_master(x: MPCTensor, y: MPCTensor, op_str: str) -> MPCTensor:
+def mul_master(x: MPCTensor, y: MPCTensor, op_str: str, kwargs_: dict) -> MPCTensor:
```
Comment (gmuraru) [inline src/sympc/protocol/spdz/spdz.py:28]: NITPICK: ```Dict[Any, Any]```

File: src/sympc/tensor/mpc_tensor.py [hunk 4]
```diff
@@ -495,33 +538,41 @@ def __apply_public_op(
         result = MPCTensor(shares=shares, session=self.session)
         return result
 
+    # TODO: put back @lru_cache(maxsize=128)
```
Comment (gmuraru) [inline src/sympc/tensor/mpc_tensor.py:541]: TODO: Don't forget :D

File: src/sympc/tensor/mpc_tensor.py [hunk 5]
```diff
@@ -495,33 +538,41 @@ def __apply_public_op(
...
-        res = op(x, y)
+        res = op(x, y, **kwargs_)
         return res.shape
 
     def __apply_op(
-        self, y: Union["MPCTensor", torch.Tensor, float, int], op_str: str
+        self,
+        y: Union["MPCTensor", torch.Tensor, float, int],
+        op_str: str,
+        kwargs_: dict = {},
```
Comment (gmuraru) [inline src/sympc/tensor/mpc_tensor.py:567]: NITPICK: Also here

**Comment by LaRiffle** [comment]
@gmuraru fixed! 😄

**Review by gmuraru — APPROVED** [review]
LGTM!

**Pull request merged**